### PR TITLE
planner: support trace for min/max eliminate

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/29661

### What is changed and how it works?

support trace for min/max eliminate

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]

File: planner/core/rule_max_min_eliminate.go [hunk 1]
```diff
@@ -222,22 +230,80 @@ func (a *maxMinEliminator) eliminateMaxMin(p LogicalPlan) LogicalPlan {
...
+			if i > 0 {
+				buffer.WriteString(",")
+			}
+			buffer.WriteString(fmt.Sprintf("%v", join.ID()))
+		}
+		buffer.WriteString(fmt.Sprintf("] as their parent during eliminate agg[%v] multi min/max functions", originAgg.ID()))
+		return buffer.String()
+	}()
+	reason := func() string {
+		buffer := bytes.NewBufferString("each column is sorted and has index in agg[")
```
Comment (time-and-fate) [inline planner/core/rule_max_min_eliminate.go:298]:
```suggestion
		buffer := bytes.NewBufferString("each column is sorted and can benefit from index/primary key in agg[")
```
Having an index is not enough to enable this optimization, so I think "benefit from" is more precise.

File: planner/core/rule_max_min_eliminate.go [hunk 2]
```diff
@@ -222,22 +230,80 @@ func (a *maxMinEliminator) eliminateMaxMin(p LogicalPlan) LogicalPlan {
...
+		if sort != nil {
+			buffer.WriteString(fmt.Sprintf("add sort[%v],", sort.ID()))
+		}
+		buffer.WriteString(fmt.Sprintf("add limit[%v] during eliminate agg[%v] %s function", limit.ID(), agg.ID(), agg.AggFuncs[0].Name))
+		return buffer.String()
+	}()
+	reason := func() string {
+		buffer := bytes.NewBufferString(fmt.Sprintf("agg[%v] has only one function[%s] without group by", agg.ID(), agg.AggFuncs[0].Name))
+		if sel != nil {
+			buffer.WriteString(fmt.Sprintf(", the columns in agg[%v] shouldn't be NULL and needs to be filer NULL out", agg.ID()))
```
[the remark this answers is not in the document]
Comment (time-and-fate) [reply planner/core/rule_max_min_eliminate.go:268]: And can we have a test case to cover this?

**Review by time-and-fate — APPROVED** [review]

File: planner/core/rule_max_min_eliminate.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Yisaer) [reply planner/core/rule_max_min_eliminate.go]: updated.

**Pull request merged**